File: Java_Version/Parcial_Java/src/main/java/parcial_java/PersonajeController.java
package parcial_java;

import db.Conexion;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class PersonajeController {

    public boolean verificarConexion() {
        Connection cn = Conexion.getConexion();
        if (cn != null) {
            try {
                cn.close();
                return true;
            } catch (SQLException e) {
                System.out.println("Error al cerrar conexion: " + e.getMessage());
            }
        }
        return false;
    }

    public boolean crearPersonaje(String nombre, int claseId, int vida) {
        String sql = "INSERT INTO personaje (nombre, clase_id, vida) VALUES (?, ?, ?)";
        try (Connection cn = Conexion.getConexion();
             PreparedStatement ps = cn.prepareStatement(sql)) {
            
            ps.setString(1, nombre);
            ps.setInt(2, claseId);
            ps.setInt(3, vida);
            
            ps.executeUpdate();
            return true;
        } catch (SQLException e) {
            System.out.println("Error al crear personaje: " + e.getMessage());
            return false;
        }
    }

    public void listarPersonajes() {
        String sql = "SELECT * FROM personaje";
        try (Connection cn = Conexion.getConexion();
             PreparedStatement ps = cn.prepareStatement(sql);
             ResultSet rs = ps.executeQuery()) {
            
            System.out.println("\n--- PERSONAJES ---");
            boolean tienePersonajes = false;
            
            while (rs.next()) {
                tienePersonajes = true;
                System.out.println("ID: " + rs.getInt("id") + 
                                   " Nombre: " + rs.getString("nombre") + 
                                   " Clase ID: " + rs.getInt("clase_id") + 
                                   " Vida: " + rs.getInt("vida"));
            }
            
            if (!tienePersonajes) {
                System.out.println("No hay personajes registrados.");
            }
        } catch (SQLException e) {
            System.out.println("Error al listar personajes: " + e.getMessage());
        }
    }
}

File: Python_Version/controlador/personaje_controller.py
from db.conexion import Conexion
from mysql.connector import Error
from modelo.personaje import Personaje, Guerrero  

class PersonajeController:
    def verificar_conexion(self):
        cn = Conexion.get_conexion()
        if cn is not None:
            try:
                cn.close()
                return True
            except Error:
                return False
        return False

    def crear_personaje(self, nombre, clase_id, vida):
        sql = "INSERT INTO personaje (nombre, clase_id, vida) VALUES (%s, %s, %s)"
        cn = Conexion.get_conexion()
        if cn is None:
            return False
        try:
            cursor = cn.cursor()
            cursor.execute(sql, (nombre, clase_id, vida))
            cn.commit()
            cursor.close()
            cn.close()
            return True
        except Error as e:
            print(f"Error al crear: {e}")
            return False

    def listar_personajes(self):
        sql = "SELECT * FROM personaje"
        cn = Conexion.get_conexion()
        if cn is None:
            return
        try:
            cursor = cn.cursor(dictionary=True)
            cursor.execute(sql)
            personajes = cursor.fetchall()
            print("\n--- LISTA DE PERSONAJES ---")
            if not personajes:
                print("No hay personajes registrados.")
            else:
                for p in personajes:
                   
                    if p['clase_id'] == 1:
                        obj_p = Guerrero(p['id'], p['nombre'], p['vida'])
                    else:
                        obj_p = Personaje(p['id'], p['nombre'], p['clase_id'], p['vida'])
                    
                    print(f"ID: {obj_p.id} | Nombre: {obj_p.nombre} (Vida: {obj_p.vida}) -> {obj_p.atacar()}")
            
            cursor.close()
            cn.close()
        except Error as e:
            print(f"Error al listar: {e}")

    def eliminar_personaje(self, id_personaje):
        sql = "DELETE FROM personaje WHERE id = %s"
        cn = Conexion.get_conexion()
        if cn is None:
            return False
        try:
            cursor = cn.cursor()
            cursor.execute(sql, (id_personaje,))
            cn.commit()
            filas = cursor.rowcount
            cursor.close()
            cn.close()
            return filas > 0
        except Error as e:
            print(f"Error al eliminar: {e}")
            return False

